### ISSUE TITLE
Cargo build seems to hang on Arch

### ISSUE THREAD

**Issue by dcellucci** [twistedfall/opencv-rust#380]
I am building opencv on an Arch VM, and I am having trouble getting the build to complete.
I don't get an error, but the build seems to hang at this point: 
```
[opencv 0.70.0] === Waiting until the binding-generator binary is built...
[opencv 0.70.0]    Compiling libc v0.2.134
[opencv 0.70.0]    Compiling glob v0.3.0
[opencv 0.70.0]    Compiling memchr v2.5.0
[opencv 0.70.0]    Compiling clang-sys v1.4.0
[opencv 0.70.0]    Compiling cfg-if v1.0.0
[opencv 0.70.0]    Compiling libloading v0.7.3
[opencv 0.70.0]    Compiling aho-corasick v0.7.19
[opencv 0.70.0]    Compiling regex-syntax v0.6.27
[opencv 0.70.0]    Compiling regex v1.6.0
[opencv 0.70.0]    Compiling clang v2.0.0
[opencv 0.70.0]    Compiling percent-encoding v2.2.0
[opencv 0.70.0]    Compiling dunce v1.0.2
[opencv 0.70.0]    Compiling once_cell v1.15.0
[opencv 0.70.0]    Compiling maplit v1.0.2
[opencv 0.70.0]    Compiling opencv-binding-generator v0.48.0
[opencv 0.70.0]     Finished release [optimized] target(s) in 1m 18s
Building [=======================>   ] 34/37: opencv(build)
```

I installed the base packages using
```
pacman -S clang qt5-base opencv
```
and I built using 
```
RUST_BACKTRACE=full cargo build -vv 
```

The full output is located here:
https://gist.github.com/dcellucci/fecbdbdaabc8c68981eef6d3c5b2d436

**Comment by twistedfall** [comment]
That's quite weird, can you please try doing a `cargo clean` beforehand and if that doesn't help try using the version `0.69.0` of the crate?

**Comment by dcellucci** [comment]
I have been running `cargo clean` each time, no help. Also downgrading to `0.69.0` didn't work.

**Comment by dcellucci** [comment]
ok after taking a closer look at the build process it seems like it's hanging on this line
```
let token = job_server.acquire().expect("Can't acquire token from job server");        
```
the thread is locked until it can acquire a token from the job server, so it appears to freeze. I will look into a solution.

UPDATE: 
After reviewing jobserver generation code, I had to manually create a job server instead of relying on `from_env()`. A manually created jobserver successfully produces tokens and the rest of the build process seems to proceed without issues.

**Comment by twistedfall** [comment]
Thank you for the investigation, that's really helpful! Can you please tell me how exactly you created the jobserver manually?

**Comment by dcellucci** [comment]
Yes - sorry for not specifying that! 

In build.rs, lines 228-238 

https://github.com/twistedfall/opencv-rust/blob/007de72d86eee96e0161bb67bb9e3b3f50fa8d7e/build.rs#L228-L238

Calling `::from_env()` returns a client, but that client always has 0 jobs. Instead creating a new client from scratch seems to work fine. I did this by modifying the function to explicitly call the `or_else` code block. Not sure if this is a jobserver problem or what.

**Comment by twistedfall** [comment]
This should be fixed in 0.74.1
